x-pack/plugins/elastic_assistant/server/routes/knowledge_base/entries/find_route.ts

```diff
@@ -64,29 +67,62 @@ export const findKnowledgeBaseEntriesRoute = (router: ElasticAssistantPluginRout
             return checkResponse;
           }
 
-          const kbDataClient = await ctx.elasticAssistant.getAIAssistantKnowledgeBaseDataClient(
-            true
-          );
+          const kbDataClient = await ctx.elasticAssistant.getAIAssistantKnowledgeBaseDataClient({
+            v2KnowledgeBaseEnabled: true,
+          });
           const currentUser = ctx.elasticAssistant.getCurrentUser();
           const userFilter = getKBUserFilter(currentUser);
           const additionalFilter = query.filter ? ` AND ${query.filter}` : '';
+          const systemFilter = ` AND NOT kb_resource:"${ESQL_RESOURCE}"`;
 
+          // TODO: Either plumb through new `findDocuments` that takes query DSL so you can do agg + pagination to collapse
+          // TODO: system entries, use scoped esClient from request, or query them separate and mess with pagination...latter for now.
```

Comment:
This is WIP and will be addressed next when moving away from `EuiInMemoryTable`.